torch/optim/adam.py

```diff
@@ -483,6 +497,15 @@ def _multi_tensor_adam(
     grouped_tensors = Optimizer._group_tensors_by_device_and_dtype(
         [params, grads, exp_avgs, exp_avg_sqs, max_exp_avg_sqs, state_steps]  # type: ignore[list-item]
     )
+
+    # We only shuffle around the beta when it is a Tensor and on CUDA, otherwise, we prefer
+    # treating it as a scalar.
+    beta1_dict: Optional[DeviceDict] = (  # type: ignore[attr-defined]
+        {beta1.device: beta1}
+        if isinstance(beta1, Tensor) and str(beta1.device) != "cpu"
+        else None
+    )
```

Comment:
How come you don't need a beta2_dict?

Reply:
Beta2 is only used in argument positions that allow the tensor to be on different devices. (ie like a scalar tensor which will automatically be moved)